src/config/Config.zig

```diff
@@ -3111,6 +3128,11 @@ pub fn finalize(self: *Config) !void {
 
     const alloc = self._arena.?.allocator();
 
+    // Ensure our launch source is properly set.
+    if (self.@"launched-from" == null) {
+        self.@"launched-from" = .detect();
+    }
```

Comment:
I dunno how I feel about this... Maybe we should add a `_launched_from` variable that is always non-null? I just don't like seeing `.?` everywhere I guess

Reply:
Yeah I felt the same way and thought the same thing, but was worried that it'd be too easy to not know which one to check as a consumer (there being 2 sources of seeming truth). The `?` is optimized away in unsafe builds and I think the safety it affords in safe builds is fine.